python/ray/dag/tests/experimental/test_torch_tensor_dag.py

```diff
@@ -886,6 +910,949 @@ def test_torch_tensor_exceptions(ray_start_regular):
     compiled_dag.teardown()
 
 
+@pytest.mark.parametrize("ray_start_regular", [{"num_cpus": 4}], indirect=True)
+def test_torch_tensor_nccl_all_reduce(ray_start_regular):
+    """
+    Test basic all-reduce.
+    """
+    if not USE_GPU:
+        pytest.skip("NCCL tests require GPUs")
+
+    assert (
+        sum(node["Resources"].get("GPU", 0) for node in ray.nodes()) > 1
+    ), "This test requires at least 2 GPUs"
+
+    actor_cls = TorchTensorWorker.options(num_cpus=0, num_gpus=1)
+
+    num_workers = 2
+    workers = [actor_cls.remote() for _ in range(num_workers)]
+
+    with InputNode() as inp:
+        computes = [
+            worker.compute_with_tuple_args.bind(inp, i)
+            for i, worker in enumerate(workers)
+        ]
+        collectives = collective.allreduce.bind(computes, ReduceOp.SUM)
+        recvs = [
+            worker.recv.bind(collective)
+            for worker, collective in zip(workers, collectives)
+        ]
+        dag = MultiOutputNode(recvs)
+
+    compiled_dag = dag.experimental_compile()
+
+    base_sum = (1 + num_workers) * num_workers / 2
+    for i in range(3):
+        i += 1
+        shape = (i * 10,)
+        dtype = torch.float16
+        ref = compiled_dag.execute(
+            [(shape, dtype, i + idx + 1) for idx in range(num_workers)]
+        )
+        result = ray.get(ref)
+        reduced_val = base_sum + num_workers * i
```

Comment:
How about just computing the expected reduced_val from the vals passed to `compiled_dag.execute`? That is easier to read. Same for following tests.

Reply:
Ack. Changed to `reduced_val = sum(i + idx for idx in range(num_workers))` where `i + idx for idx in range(num_workers)` are the vals passed to `compiled_dag.execute`.